File: review-engine/examples/rule_self_tests/cases/java/shared_security/violating.java
class SecurityReview {
  private static final String apiKey = "sk_live_java_fixture";

  void loadUser(Connection conn, String userId, String table) throws Exception {
    Runtime.getRuntime().exec("id " + userId);
    String query = "SELECT * FROM " + table + " WHERE id = " + userId;
    conn.createStatement().executeQuery(query);
  }
}

File: review-engine/examples/rule_self_tests/cases/typescript/shared_security/violating.ts
import * as child_process from "node:child_process";

export function loadUser(db: Database, userId: string, table: string) {
  const token = "sk_live_typescript_fixture";
  child_process.exec("id " + userId);
  const query = `SELECT * FROM ${table} WHERE id = ${userId}`;
  return db.query(query);
}
